src/app/core/admin/admin-entity-list/admin-entity-list.component.html

```diff
@@ -61,31 +59,31 @@
         [appAdminTabTemplate]="config?.columnGroups?.groups"
         let-item
       >
-        <div
-          class="table-header drop-list flex-row"
-          cdkDropList
-          cdkDropListOrientation="horizontal"
-          cdkDropListLockAxis="x"
-          (cdkDropListDropped)="drop($event, item.columns)"
-        >
-          @for (col of item["columns"]; track col) {
-            <div cdkDrag class="column-header drop-item flex-row gap-small">
-              <fa-icon icon="grip-vertical" class="drag-handle"></fa-icon>
+        <app-reorderable-list
+          [items]="item.columns"
+          [orientation]="'horizontal'"
+          [dropLockAxis]="'x'"
+          [itemTemplate]="columnTemplate"
+          (itemsChange)="item.columns = $event"
+        ></app-reorderable-list>
 
-              <!-- TODO: for custom column overrides, this cannot display the label yet -->
-              <app-entity-field-label
-                [field]="col"
-                [entityType]="entityConstructor"
-              ></app-entity-field-label>
+        <ng-template #columnTemplate let-col let-remove="remove">
+          <div class="column-header drop-item flex-row gap-small">
+            <fa-icon icon="grip-vertical" class="drag-handle"></fa-icon>
 
-              <fa-icon
-                icon="times"
-                class="remove-icon"
-                (click)="item.columns = removeItem(item.columns, col)"
-              ></fa-icon>
-            </div>
-          }
-        </div>
+            <!-- TODO: for custom column overrides, this cannot display the label yet -->
+            <app-entity-field-label
+              [field]="col"
+              [entityType]="entityConstructor"
+            ></app-entity-field-label>
+
+            <fa-icon
+              icon="times"
+              class="remove-icon"
+              (click)="remove(col)"
+            ></fa-icon>
+          </div>
+        </ng-template>
 
         <div class="table-content-preview">
           <span i18n>Drag & drop table column headers. Add columns:</span>
```

Comment:
I am wondering where we should cut the components :thinking: 

Is there a use case to only use the drag-and-drop-columns without an option to add columns? Or should we put the the entity-fields-menu inside that also, to have a rather large component that can administer columns completely with an [(columns)] input+output?

Reply:
I tried this out and found a few issues:

In the AdminEntityList, we were using drag-and-drop for both filters and columns. However, filters are of type string[], while columns are of type ColumnConfig[]. Because of this difference, we need to handle drag-and-drop, as well as field addition/removal, separately for each.

also, I noticed that the current drag-and-drop implementation (in our new component) is not working correctly for columns in the list view.


just wanted to let you know